[Segment Replication] SearchableSnapshotIT class tests are failing with Segment Replication enabled Indices

**Describe the bug**
Currently SearchableSnapshotIT class tests are failing with Segment Replication enabled Indices. The failure happens when we are restoring snapshot with storage type as `REMOTE_SNAPSHOT`. The possible cause for this failure is [here](https://github.com/opensearch-project/OpenSearch/blob/8b34e5f724fe2b875731228795f51d523462599a/server/src/main/java/org/opensearch/indices/IndicesService.java#L933), as we are wiring up a `NRTReplicationEngineFactor` instead of `ReadOnlyEngine` when the index is `REMOTE_SNAPSHOT`.

```
java.lang.AssertionError: timed out waiting for green state
	at __randomizedtesting.SeedInfo.seed([7CAA2E07C2E84004:2BEC37D99F2F6F46]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.opensearch.test.OpenSearchIntegTestCase.ensureColor(OpenSearchIntegTestCase.java:1007)
	at org.opensearch.test.OpenSearchIntegTestCase.ensureGreen(OpenSearchIntegTestCase.java:938)
	at org.opensearch.test.OpenSearchIntegTestCase.ensureGreen(OpenSearchIntegTestCase.java:927)
	at org.opensearch.snapshots.SearchableSnapshotIT.restoreSnapshotAndEnsureGreen(SearchableSnapshotIT.java:324)
	at org.opensearch.snapshots.SearchableSnapshotIT.testCreateSearchableSnapshotWithChunks(SearchableSnapshotIT.java:136)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
```

**To Reproduce**
Steps to reproduce the behavior:
Set Segment replication enabled for indices and then run the  SearchableSnapshotIT test suite.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Plugins**
Please list all plugins currently enabled.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/server/src/main/java/org/opensearch/indices/IndicesService.java
+++ b/server/src/main/java/org/opensearch/indices/IndicesService.java
@@ -930,11 +930,11 @@ public class IndicesService extends AbstractLifecycleComponent
             .filter(maybe -> Objects.requireNonNull(maybe).isPresent())
             .collect(Collectors.toList());
         if (engineFactories.isEmpty()) {
-            if (idxSettings.isSegRepEnabled()) {
-                return new NRTReplicationEngineFactory();
-            }
             if (idxSettings.isRemoteSnapshot()) {
                 return config -> new ReadOnlyEngine(config, new SeqNoStats(0, 0, 0), new TranslogStats(), true, Function.identity(), false);
+            }
+            if (idxSettings.isSegRepEnabled()) {
+                return new NRTReplicationEngineFactory();
             }
             return new InternalEngineFactory();
         } else if (engineFactories.size() == 1) {

